src/python/gem5/components/boards/riscv_board.py

```diff
@@ -541,6 +541,17 @@ def _pre_instantiate(self, full_system: Optional[bool] = None):
             else:
                 self.workload.kernel_addr = 0x0
                 self.workload.entry_point = 0x80000000
+
+            # Set up the device tree. We need to wait until pre-instantiate to
+            # do this since the workload could change until this point.
+            # Default DTB address if bbl is built with --with-dts option
+            self.workload.dtb_addr = 0x87E00000
+
+            self.generate_device_tree(m5.options.outdir)
+            self.workload.dtb_filename = os.path.join(
+                m5.options.outdir, "device.dtb"
+            )
```

Comment:
Shouldn't this only ben run for FS mode?

Reply:
Isn't it?

Reply:
Yip! I misread the change.